glean-core/ios/Glean/Dispatchers.swift

```diff
@@ -43,39 +51,27 @@ class Dispatchers {
     // property
     private init() {}
 
-    /// This function launches a background `Operation`
+    /// This function launches a background `Operation` on a serially executed queue.
     ///
     /// - parameters:
     ///   * block: The block of code to execute, as a Closure that accepts no arugments and returns Void
     ///
     /// This function is used throughout Glean in order to launch tasks in the background, typically
-    /// recording of metrics, uploading of pings, etc.  This function returns an `Operation` which can
-    /// be used in conjunction with a completion handler to perform work after the task is complete.
-    ///
-    /// For example:
-    /// ```
-    /// Dispatchers.shared.launch {
-    ///     // Do some work here
-    /// }.completionBlock = {
-    ///     print("Done!")
-    /// }
-    /// ```
-    ///
-    /// You could also use the returned `Operation` to await that specific task like this:
-    /// ```
-    /// let operation = Dispatchers.shared.launch {
-    ///     // Do some work here
-    /// }
-    /// operation.waitUntilFinished()
-    func launch(block: @escaping () -> Void) -> Operation {
-        let operation = GleanOperation(block)
-
+    /// recording of metrics and things that need to execute in order. Since this executes the tasks on
+    /// a non-concurrent (serial) queue, the tasks are executed in the order that they are launched.
+    ///
+    /// If `queueInitialTasks` is enabled, then the operation will be created and added to the
+    /// `preInitOperations` array but not executed until flushed.
+    ///
+    /// If `testingMode` is enabled, then launch will await the execution of the task (unless queuing is
+    /// enabled)
+    func launch(block: @escaping () -> Void) {
```

Comment:
It's async (as in it happens on a background thread), but the execution of that particular thread queue is serial rather than concurrent so only one task launched by this function can be executed at a time.  I think the naming of the `launchAsync` functions I added is probably the problem here, they should be `launchConcurrent`.  I intended to have something that we could use that would be like running a task in a coroutine.

Reply:
Ok, I thought so. I think we should really hint that this is used for the API. We don't want other stuff going on this queue. I'd also drop the async suffix, as everything is async here

Reply:
I'll rename these to be more descriptive.  `launchAPI` for the serial queue, and I can rename the other to `launchConcurrent` or similar.